x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/FieldPermissionsCache.java

```diff
@@ -41,13 +44,37 @@ public final class FieldPermissionsCache {
     );
     private final Cache<FieldPermissionsDefinition, FieldPermissions> cache;
 
+    public static FieldPermissionsCache init(Settings settings) {
+        final FieldPermissionsCache cache = new FieldPermissionsCache(settings);
+        INSTANCE.set(cache);
+        return cache;
```

Comment:
It feels a bit odd to have both a staticly shared instance and a public constructor.